### PR TITLE
chore: Top-level Makefile

### PR DESCRIPTION
**Pull request by jefft0**
This pull request has 3 commits. The first commit fixes the build of the iOS Example app: Unwrap the result of sendToDict which is now optional. Update the IPHONEOS_DEPLOYMENT_TARGET to 13.

The second commit simply moves files from the packages folder to the top level as-is. We do this so that the next commit will show the changes to the Makefile. Also, we move `deps.mod` to the go folder and we delete packages/go.mod.

The third commit updates the Makefile with the correct paths for `GO_DIR`, etc. We also use `cd $(GO_DIR) && ` before every go command, which allows us to use a single `go.mod` file in the go folder. In the `go.mod` file we merge entries from `packages/go.mod` (now deleted). Finally, in Android, iOS and workflow scripts we update paths with /packages to the top folder.

### CODE REVIEW

**Comment by jefft0** [comment]
Hello @aeddi and @gfanton . This will become the pull request to move the Makefile from packages to the top level. Before I move it, I wanted the build for the iOS Example app to work (so that I can check if it still builds after moving the Makefile). Previously, the Xcode project linked to GomobileIPFS.framework, but it seems like the Makefile doesn't build this. Should it? (In this pull request I changed the Example app to compile the Swift files for GomobileIPFS directly.)

File: .github/workflows/build-ios.yml [hunk 1]
```diff
@@ -57,35 +57,35 @@ jobs:
           pod setup
 
       - name: Build iOS packages
-        working-directory: packages
+        working-directory: .
```
Comment (aeddi) [inline .github/workflows/build-ios.yml:60]: You can remove this line

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -127,7 +127,7 @@ jobs:
           java-version: 11
 
       - name: Build Go Core
-        working-directory: packages
+        working-directory: .
```
Comment (aeddi) [inline .github/workflows/lint.yml:130]: You remove this line

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -155,7 +155,7 @@ jobs:
 
       - name: Build iOS artifacts
         if: steps.semantic.outputs.new_release_published == 'true'
-        working-directory: packages
+        working-directory: .
```
Comment (aeddi) [inline .github/workflows/release.yml:158]: You can remove this line

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -166,15 +166,15 @@ jobs:
...
+            build/ios/cocoapods/core/${{ steps.semantic.outputs.new_release_version }}/Gomobile-IPFS-Core-v${{ steps.semantic.outputs.new_release_version }}.pod.zip
+            build/ios/cocoapods/core/${{ steps.semantic.outputs.new_release_version }}/Gomobile-IPFS-Core.podspec
+            build/ios/cocoapods/bridge/${{ steps.semantic.outputs.new_release_version }}/Gomobile-IPFS-Bridge-v${{ steps.semantic.outputs.new_release_version }}.pod.zip
+            build/ios/cocoapods/bridge/${{ steps.semantic.outputs.new_release_version }}/Gomobile-IPFS-Bridge.podspec
           # yamllint enable rule:line-length
 
       - name: Publish iOS packages
         if: steps.semantic.outputs.new_release_published == 'true'
-        working-directory: packages
+        working-directory: .
```
Comment (aeddi) [inline .github/workflows/release.yml:177]: You can remove this line

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -178,7 +178,7 @@ jobs:
             -destination "platform=iOS Simulator,id=$DESTINATION"
 
       - name: Build iOS Bridge
-        working-directory: packages
+        working-directory: .
```
Comment (aeddi) [inline .github/workflows/test.yml:181]: You can remove this line

File: ios/Example/Example.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -147,7 +146,6 @@
 		8DAEFB1A23FF38F300818B81 /* Frameworks */ = {
 			isa = PBXGroup;
 			children = (
-				8AF76E4426E23ED4007DF24A /* Core.xcframework */,
```
Comment (aeddi) [inline ios/Example/Example.xcodeproj/project.pbxproj:150]: This change looks really weird. Same thing goes for the change line 59. Looks like you removed the reference to the Core.xframework ?

Comment (aeddi) [reply ios/Example/Example.xcodeproj/project.pbxproj:150]: Ha ok I see. Indeed it must have been a mistake made by d4ryl00.
No problem then, you can keep this change. 👍

**Review by aeddi — APPROVED** [review]
LGTM 👍

**Pull request merged**